Sync Community PR #2041 ([v2.13.1] Rancher IPv6 Support) (#619)
rancher/rancher-product-docs: versions/v2.13/modules/en/pages/cluster-deployment/dual-stack.adoc

= IPv4/IPv6 Dual-stack
ifeval::["{build-type}" != "srfa"]
:page-languages: [en, zh]
endif::[]
:revdate: 2025-12-10
:page-revdate: {revdate}

Kubernetes supports IPv4-only, IPv6-only, and dual-stack networking configurations.
For more details, refer to the official https://kubernetes.io/docs/concepts/services-networking/dual-stack/[Kubernetes documentation].

== Installing Rancher on IPv6-Only or Dual-Stack Clusters

Rancher can run on clusters using:

* IPv4-only
* IPv6-only
* Dual-stack (IPv4 + IPv6)

When you install Rancher on an *IPv6-only cluster*, it can communicate externally *only over IPv6*. This means it can provision:

* IPv6-only clusters
* Dual-stack clusters  

>   **(IPv4-only downstream clusters are not possible in this case)**

Reviewer: ```suggestion
  _(IPv4-only downstream clusters are not possible in this case)_
```

In the original PR this was italicized and not bolded. If this change is intention, please ignore.

Author: Unintentional, updated in https://github.com/rancher/rancher-product-docs/pull/619/commits/c47bbc969c0a115442b8bcf9430b0bd956940b61.

Note that Windows installations of RKE2 do not support dual-stack clusters using BGP.
For more details, see https://documentation.suse.com/cloudnative/rke2/latest/en/networking/basic_network_options.html[RKE2 Network Options].

=== Provisioning Node Driver Clusters

Rancher currently supports assigning IPv6 addresses in *node driver* clusters with:

* xref:cluster-deployment/infra-providers/aws/machine-configuration.adoc[Amazon EC2]
* xref:cluster-deployment/infra-providers/digitalocean/machine-configuration.adoc[DigitalOcean]

Support for additional providers will be introduced in future releases.

[NOTE]
.DigitalOcean Limitation
====
Creating an **IPv6-only cluster** using the DigitalOcean node driver is currently **not supported**. For more details, please see https://github.com/rancher/rancher/issues/52523#issuecomment-3457803572[rancher/rancher#52523].
====

==== Infrastructure Requirements

Cluster nodes must meet the requirements listed in the xref:cluster/deployment/node-requirements.adoc[Node Requirements for Rancher Managed Clusters].

Machine pool configuration guides:

* xref:cluster-deployment/infra-providers/aws/machine-configuration.adoc[Amazon EC2]

> * xref:cluster-deployment/infra-providers/digitalocean/machine-configuration.adoc[DigitalOcean]

Reviewer: ```suggestion
* xref:cluster-deployment/infra-providers/aws/aws.adoc[Amazon EC2]
* xref:cluster-deployment/infra-providers/digitalocean/digitalocean.adoc[DigitalOcean]
```

=== Provisioning Custom Clusters

To provision on your own nodes, follow the instructions in xref:cluster-deployment/custom-clusters/custom-clusters.adoc[Provision Kubernetes on Existing Nodes].

[NOTE]
====
* *Node Public IP* and *Node Private IP* fields accept IPv4, IPv6, or both (comma-separated).  
    >   Example: `10.0.0.5,2001:db8::1`
* In *IPv6-only* and *dual-stack* clusters, specify the node’s *IPv6 address* as the *Private IP*.
====

==== Infrastructure Requirements

Infrastructure requirements are the same as above for node-driver clusters.

== Other Limitations

=== GitHub.com

GitHub.com does *not* support IPv6. As a result:

> * Any application repositories ( `ClusterRepo.catalog.cattle.io/v1` CR) hosted on GitHub.com will *not be reachable* from IPv6-only clusters.

Reviewer: ```suggestion
* Any application repositories ( `ClusterRepo.catalog.cattle.io/v1` CR) hosted on GitHub.com are *not reachable* from IPv6-only clusters.
```

> * Similarly, any *non-builtin node drivers* hosted on GitHub.com will also *not be accessible* in IPv6-only environments.

Reviewer: ```suggestion
* Similarly, any *non-builtin node drivers* hosted on GitHub.com are also *not accessible* in IPv6-only environments.
```
